### ISSUE TITLE
KVStore corrupts binary data

### ISSUE THREAD

**Issue by eminence** [cloudflare/workers-rs#101]
I'm trying to store binary data in the KV, but the data appears to getting corrupted.  Consider this code:

```rust
.get_async("/test", |req, ctx| async move {
            let kv = ctx.kv("UPLOAD")?;
            let bytes = [0xd2, 0xda, 0x1e, 0x77, 0x3c, 0x50, 0xd9, 0x35, 0x9a, 0x89, 0xd5, 0x53, 0xc3, 0x81, 0x6d, 0x20];
            kv.put_bytes("test", &bytes)?.execute().await?;

            let bytes_from_kv = kv.get("test").await?.unwrap();
            assert_eq!(&bytes, bytes_from_kv.as_bytes());

            Response::empty()
        })
```

Trying to run this yields this assertion failure:

```
panicked at 'assertion failed: `(left == right)`
  left: `[210, 218, 30, 119, 60, 80, 217, 53, 154, 137, 213, 83, 195, 129, 109, 32]`,
 right: `[239, 191, 189, 239, 191, 189, 30, 119, 60, 80, 239, 191, 189, 53, 239, 191, 189, 239, 191, 189, 239, 191, 189, 83, 195, 129, 109, 32]`', src/lib.rs:56:13
```

However when I download this value via the https://dash.cloudflare.com/, the data is correct (so it seems like the corruption is happening during the fetch on the rust side)

## Version info
* worker-rs version v0.0.7
* wrangler version 1.19.6
* rustc version 1.59.0-nightly (7abab1efb 2021-12-17)

**Comment by nilslice** [comment]
Thanks for filing this - @zebp, any chance you can take a look?

**Comment by zebp** [comment]
Hey @eminence! This was fixed already in a currently unreleased version of my [worker-kv](https://github.com/zebp/worker-kv) crate. Sadly this version contains necessary breaking changes to reach parity with the JS KV api. Simply specify `worker-kv` manually with:
```toml
[dependencies]
worker-kv = { git = "https://github.com/zebp/worker-kv" }
```

and this is the route handler I used:

```rust
let kv = worker_kv::KvStore::from_this(&ctx.get_env(), "UPLOAD").unwrap();
let bytes = [
    0xd2, 0xda, 0x1e, 0x77, 0x3c, 0x50, 0xd9, 0x35, 0x9a, 0x89, 0xd5, 0x53, 0xc3, 0x81,
    0x6d, 0x20,
];
kv.put_bytes("test", &bytes)
    .unwrap()
    .execute()
    .await
    .unwrap();

let bytes_from_kv = kv.get("test").bytes().await.unwrap().unwrap();
assert_eq!(&bytes, bytes_from_kv.as_slice());

Response::empty()
```

**Comment by nilslice** [comment]
Thanks, @zebp! Do you think you'll want to cut a release for your crate soon? I think we're about ready to get a 0.0.8 workers-rs released, but could also definitely consider a breaking change and bump the minor version.

**Comment by eminence** [comment]
Thanks @zebp, I'll give this a try.  It would be nice to have this as part of a release at some point.

IMHO, a breaking API change is justified to fix a silent data corruption bug, especially in a `0.0.x` project (where breaking changes are allowed at any time)

**Comment by eminence** [comment]
I confirm your latest version works @zebp , thanks.  Unfortunately mixing in your version is not ideal since it's not possible to use the question-mark operator to propagate errors:

```
error[E0277]: the trait bound `worker::Error: From<worker_kv::KvError>` is not satisfied
the trait `From<worker_kv::KvError>` is not implemented for `worker::Error`
```

So I'm definitely looking forward to seeing this fully integrated into the next `workers-rs` version 👍

**Comment by zebp** [comment]
> Thanks, @zebp! Do you think you'll want to cut a release for your crate soon? I think we're about ready to get a 0.0.8 workers-rs released, but could also definitely consider a breaking change and bump the minor version.

I was going to wait for the inline-js issues to be fully resolved so I implement streaming, but I guess this issue seems important enough for `worker-kv` to get a release. I'll do that tonight or tomorrow.

**Comment by nilslice** [comment]
Ok cool, thank you! Re: inline-js, I need to re-visit where it was left off. Maybe we can get both in for the next release...

**Comment by zebp** [comment]
I just released worker-kv [0.5.0](https://crates.io/crates/worker-kv/0.5.0) and created #103 updating the dependency.
